status/condition_set.go

```diff
@@ -237,3 +240,62 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// unstructuredAdapter is an embedded struct type that embeds the upstream type. unstructuredAdapter
+// makes the assumption that status conditions are found on status.conditions path.
+type unstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+// FromUnstructured makes the assumption that the status conditions are found on status.conditions
+// path. If they are not found then we return nil.
+func FromUnstructured(u *unstructured.Unstructured) Object {
+	_, found, err := unstructured.NestedSlice(u.Object, "status", "conditions")
+	if err != nil || !found {
+		return nil
+	}
+	return &unstructuredAdapter{Unstructured: u}
+}
+
+func (u *unstructuredAdapter) GetConditions() []Condition {
+	var conditions []Condition
+	c, _, _ := unstructured.NestedSlice(u.Object, "status", "conditions")
+	for _, condition := range c {
+		var newCondition Condition
+		cond := condition.(map[string]interface{})
+		jsonStr, err := json.Marshal(cond)
```

Comment:
Should we perform validation on these things ahead of time in the constructor path so that it's impossible for us to fail on these methods inside of the interface here. Silent failure like this is dubious because you drop data without telling the user why you have dropped the data.

We can't return an error here because we wouldn't be implementing the interface if we did, so we should return an error when constructing by performing pre-validation on the data that we are manipulating

Reply:
Agreed :thumbsup: